compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -846,6 +846,16 @@ impl<'ll, 'tcx> MiscCodegenMethods<'tcx> for CodegenCx<'ll, 'tcx> {
         attributes::apply_to_llfn(llfn, llvm::AttributePlace::Function, &attrs);
     }
 
+    fn apply_target_features(&self, llfn: &'ll Value) {
+        let mut attrs = SmallVec::<[_; 2]>::new();
+        let global_features = self.tcx.global_backend_features(()).iter().map(|s| s.as_str());
+        let target_features: String = global_features.intersperse(",").collect();
+        if !target_features.is_empty() {
+            attrs.push(llvm::CreateAttrStringValue(self.llcx, "target-features", &target_features));
+        }
+        attributes::apply_to_llfn(llfn, llvm::AttributePlace::Function, &attrs);
+    }
```

Comment:
Looks like this was copied from `fn llfn_attrs_from_instance`, which lives in `rustc_codegen_llvm`, and is used only in `rustc_codegen_llvm`.
So I'd suggest to 1) keep this inside llvm codegen and do not introduce a new interface to codegen ssa, and 2) reuse the code in both `llfn_attrs_from_instance` and `create_entry_fn` without copypaste.